solarforecastarbiter/reference_forecasts/forecast.py

```diff
@@ -213,10 +216,40 @@ def unmix_intervals(cloud_cover):
 
     Parameters
     ----------
-    cloud_cover : pd.Series
+    data : pd.Series
+        The first time must be the first output of a cycle.
 
     Returns
     -------
     pd.Series
+        Data is the unmixed interval average with ending label.
     """
-    raise NotImplementedError
+    intervals = (mixed.index[1:] - mixed.index[:-1]).unique()
+    if len(intervals) > 1:
+        raise ValueError('multiple interval lengths detected. slice forecasts '
+                         'into sections with unique interval lengths first.')
+    interval = intervals[0]
+    mixed_vals = np.array(mixed)
+    if interval == pd.Timedelta('1h'):
```

Comment:
I think some comments here about exactly what this will be helpful

Reply:
is it not clear from the doc string?

Reply:
It just took me a few minutes to understand what mixed intervals are and that your math is right. An inline comment like 
``mixed_3 = avg 0-3 = (0-1 + 1-2 + 1-3) /3 => avg 1-3 = 3 * mixed3 - (0-1 + 1-2) = 3 * mixed_3 - 2 * mixed_2``
would help I think

Reply:
hopefully makes more sense now.

Reply:
yes